src/serializer/ResidualFunctions.js

```diff
@@ -439,13 +439,7 @@ export class ResidualFunctions {
 
         factoryParams = factoryParams.concat(params).slice();
 
-        // The Replacer below mutates the AST, so let's clone the original AST to avoid modifying it
-        let factoryNode = t.functionExpression(
-          null,
-          factoryParams,
-          ((t.cloneDeep(funcBody): any): BabelNodeBlockStatement)
-        );
-
+        let factoryNode = t.functionExpression(null, factoryParams, funcBody);
```

Comment:
I think we still need to clone, in case the body is also referenced nested in another residual function.